pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -738,7 +737,7 @@ func (c *Cluster) getActivateOSDInitContainer(configDir, namespace, osdID string
 			Path: filepath.Join(
 				configDir,
 				namespace,
-				strings.ReplaceAll(osdInfo.BlockPath, "/", "_"),
+				c.clusterInfo.FSID+"_"+osdInfo.UUID,
```

Comment:
I'm wondering if the underscore shouldn't be a `-` instead, it feels more common for a directory name, what do you think?

Reply:
It's intentional. It is to disrinct cluster fsid and osd id. If we use "-", the boundary will be unclear since these ids include many "-" like XXXX-YYYY-ZZZ-...

Reply:
Will this have any impact on upgraded OSDs? Does the OSD not rely on a consistent host path?

Reply:
There is just one concern. If the osd crashes during upgrading, the old pod and the new upgrade osd points different file locks. However, it's a very rare case and it's far dangerous to keep the current implementation as is.

Reply:
Ok, that's just the same risk during upgrade as we already had if it crashed. Then after the upgrade it would be fixed. Sounds good in that case. Were you able to confirm in a manual upgrade?

Reply:
Yes. I confirmed upgrade process worked fine. I reconfirmed as follows.

1. Created a simple Rook(v1.6.2)/Ceph host-based cluster. It consists of 1 OSD backed by /dev/sdb.

2. Cluster worked fine and its directory is _dev_sdb. It's the old behavior.

```
$ ls /var/lib/rook/rook-ceph/
client.admin.keyring  crash  _dev_sdb  log  rook-ceph.config
```

3. Upgraded to my binary that is master + this PR and it still worked fine. 

```
$ ls /var/lib/rook/rook-ceph/
39243739-0ffa-4452-b7fa-7c42e67b4268_4a4ba7f1-82f0-4629-9477-5382864129b7  client.admin.keyring  crash  _dev_sdb  log  rook-ceph.config
```

`39243739-0ffa-4452-b7fa-7c42e67b4268_4a4ba7f1-82f0-4629-9477-5382864129b7` is the new directory. _dev_sdb is a garbage, but is not harmless and is small.

4. After deleting OSD pod, this cluster looked fine.

Reply:
Great, thanks for the detailed confirmation!